tar.go

```diff
@@ -29,11 +30,13 @@ func NewTarStreamer(r io.Reader, keywords []string) Streamer {
 		tarReader:  tar.NewReader(pR),
 		keywords:   keywords,
 	}
-	go ts.readHeaders() // I don't like this
+
+	go ts.readHeaders()
```

Comment:
> When is it that you would have both? I don't think that ought to happen.

It _shouldn't_ happen. But if we want to make it an explicit decision then we need to enforce the mutual exclusivity of the keys inside `parse.go` so that I can make the comparison code nicer inside `compare.go`.